lib/models/application.coffee

```diff
@@ -376,34 +376,51 @@ getApplicationModel = (deps, opts) ->
 	# @function
 	# @memberof resin.models.application
 	#
-	# @param {String} name - application name
-	# @param {String} deviceType - device type slug
-	# @param {(Number|String)} [parentNameOrId] - parent application name or id
+	# @param {Object} options - application creation parameters
+	# @param {String} options.name - application name
+	# @param {(Number|String)} options.applicationType - application type slug or id
+	# @param {String} options.deviceType - device type slug
+	# @param {(Number|String)} [options.parent] - parent application name or id
 	#
 	# @fulfil {Object} - application
 	# @returns {Promise}
 	#
 	# @example
-	# resin.models.application.create('My App', 'raspberry-pi').then(function(application) {
+	# resin.models.application.create({ name: 'My App', applicationType: 'essentials', deviceType: 'raspberry-pi').then(function(application) {
 	# 	console.log(application);
 	# });
 	#
 	# @example
-	# resin.models.application.create('My App', 'raspberry-pi', 'ParentApp').then(function(application) {
+	# resin.models.application.create({ name: 'My App', applicationType: 'microservice', deviceType: 'raspberry-pi', parent: 'ParentApp' }).then(function(application) {
 	# 	console.log(application);
 	# });
 	#
 	# @example
-	# resin.models.application.create('My App', 'raspberry-pi', function(error, application) {
+	# resin.models.application.create({ name: 'My App', applicationType: 'microservice-starter', deviceType: 'raspberry-pi' }, function(error, application) {
 	# 	if (error) throw error;
 	# 	console.log(application);
 	# });
 	###
-	exports.create = (name, deviceType, parentNameOrId, callback) ->
+	exports.create = ({ name, applicationType, deviceType, parent }, callback) ->
 		callback = findCallback(arguments)
 
-		parentAppPromise = if parentNameOrId
-			exports.get(parentNameOrId, $select: [ 'id' ])
+		applicationTypePromise = if isId(applicationType)
+			Promise.resolve(applicationType)
```

Comment:
Let's simplify a tiny touch: even in our own code we use application types instead of ids, so let's just not support application type ids at all, and drop this case. If you want an app type, use the slug.